pyproject.toml

```diff
@@ -32,16 +31,20 @@ keywords = [
 ]
 dependencies = [
   "anywidget>=0.9.0",
+  "fast_hdbscan",
+  "geoindex-rs",
   "ipywidgets>=7.6,<9",
   "ipython",
-  "matplotlib",
+  "matplotlib>=3.9",
+  "numba",
   "numpy",
   "pandas",
   "pooch>=1.3.0",
   "pyarrow",
+  "pycolormap-2d",
   "scipy",
   "seaborn",
-  "traittypes>=0.2.1",
+  "tqdm",
```

Comment:
Not a huge deal, but I'm wondering if there is a way to feature flag these additional dependencies with an `project.optional-dependencies`. Something like:

```sh
pip install "jupyter-scatter[all]"
```

or 

```sh
pip install "jupyter-scatter[labels]"
```

and then have a runtime error if someone enters a code path with a dependency that is required for a specific feature.

Reply:
I'm all for it. I think almost all labeling features work without any new additions other than the geoindex which is lightweight.

Do you know how we would raise errors appropriately for features that would require optional dependencies?